compiler/compile.bal

```diff
@@ -72,7 +72,13 @@ function compileBalFile(string filename, string basename, string? outputBasename
 
 function processModule(CompileContext cx, bir:ModuleId id, front:SourcePart[] sourceParts, string? outFilename) returns front:ResolvedModule|CompileError {
     front:ScannedModule scanned = check front:scanModule(sourceParts, id);
-    ResolvedImport[] resolvedImports = from var mod in scanned.getImports() select check resolveImport(cx, mod);
+    // Fallowing doesn't properly pass the error back to the calling function if we get an error the import
+    // ResolvedImport[] resolvedImports = from var mod in scanned.getImports() select check resolveImport(cx, mod);
+    ResolvedImport[] resolvedImports = [];
```

Comment:
The new code looks equivalent to me. When you say `doesn't properly pass back the error` do you mean the code is wrong or that there's a JBUG?

Reply:
What I meant was check doesn't seems to return the error so that I am not getting the errors in imports back to `testCompileEU`

Reply:
That should have a JBUG comment then.  But I'm surprised because in other cases of this kind of jBallerina bug, I found the generated code crashed. When resolveImport is returning an error, what is happening?

Reply:
If we have it in this way when we hit something like an semantic error since we have checks in all the function calls it get passed all the way back to `testCompileEU` where we handle the error to make sure it is what we are expecting. But if we have it in the previous way it get test failure saying 
```
 '+' not supported for operand types {"location":{"filename":"bar-e.bal","startPos":null,"endPos":null},"functionName":"b"}
at wso2.nballerina.err.0_1_0:semantic(err.bal:61)
```
and stack trace ends at 
```
wso2.nballerina.0_1_0.tests.testSuite:testCompileEU(tests/testSuite.bal:42)
```
(that line was  `CompileError? err = testCompileFile(path);`)

But I am not certain it's a JBUG because so far I when I tried to create an isolated test case it has worked correctly

Reply:
Hmm, I tried to see if there is a difference in behavior, I didn't come across any. This is what I tried https://gist.github.com/manuranga/7ff76b633dd38c0e23b5fb61c5701c96 

Please see if you can modify above to give different behaviors under `iter` vs `query`.

Reply:
I couldn't fine a reported issue in jBallerina. I tried to replicate the issue by returning a fake error in `resolveImport`, but an error got return as expected.
So still unclear on to replicate, or even if there is an issue in the first place.

@gimantha can you please have a call with @heshanpadmasiri and figure out if something wrong is going on here, and create an issue if there is.

Reply:
Sure will have a look